pkg/alertmanager/operator.go

```diff
@@ -1667,6 +1678,36 @@ func (c *Operator) createOrUpdateWebConfigSecret(ctx context.Context, a *monitor
 	return nil
 }
 
+func (c *Operator) createOrUpdateClusterTLSConfigSecret(ctx context.Context, a *monitoringv1.Alertmanager) error {
+	var fields monitoringv1.ClusterTLSConfigFields
+	if a.Spec.ClusterTLSConfig != nil {
```

Comment:
(suggestion) we could also remove the secret if a.Spec.ClusterTLSConfig is nil to be consistent with makeStatefulSetSpec().

Reply:
Maybe we can do this in a follow up PR?

Reply:
Also, just a reminder that we still mount something like:
```Go
{
					Name:             "cluster-tls-config",
					ReadOnly:         true,
					MountPath:        "/etc/prometheus/cluster_tls_config/cluster-tls-config.yaml",
					SubPath:          "cluster-tls-config.yaml",
					MountPropagation: nil,
					SubPathExpr:      "",
				},
```
When the cluster TLS hasn't been defined. We might want to revisit this.